### DISCUSSION TITLE
AnyFixture with multiple databases

### DISCUSSION THREAD

**Discussion by hvenables** [test-prof/test-prof#239] [Ideas]
Hey,

Quick question around the use of any fixture with multiple databases. I found that when i started using it for our secondary database that the fixtures weren't getting rolled back. The clean method https://github.com/test-prof/test-prof/blob/fd399f43721d0c7ca436b570844166b621bb2701/lib/test_prof/any_fixture.rb#L166 only seems to roll back ActiveRecord::Base and there is no config option to get it to rollback a different database.

Do you have any advice on using any fixture with a secondary database?

Thanks

Harry :)

**Comment by palkan** [comment]
Hey!

Yeah, AnyFixture is not multi-db aware yet.

In order to fix this, we need to:
- Keep track of not only affected tables but the corresponding database configurations.
- Clean all the affected configurations (see https://github.com/rails/rails/blob/b764fc24c88fbd7caf580d8ac3e94e3aaccd2726/activerecord/lib/active_record/test_fixtures.rb#L163-L167). 

Thus, that's something, that should be implemented within AnyFixture.

Another useful feature would be `before_reset` and `after_reset` callbacks, so users could enhance the existing functionality.
